cosmosapi/client.go

```diff
@@ -90,7 +90,11 @@ func (c *Client) query(ctx context.Context, link string, body, ret interface{},
 }
 
 func (c *Client) method(ctx context.Context, method, link string, ret interface{}, body io.Reader, headers map[string]string) (*http.Response, error) {
-	req, err := http.NewRequest(method, path.Join(c.Url, link), body)
+	u, err := url.Parse(strings.Join(append([]string{c.Url}, link), "/"))
```

Comment:
What is the bug? Why is path.Join not sufficient? Perhaps we should have a testcase for this since it is not obvious to me (nor Alex and Johannes obviously)?